### ISSUE TITLE
[logic]: jesus rooster

### ISSUE THREAD

**Issue by foxsouns** [daid/LADXR#84]
not sure if anybody else knows this, but you can escape an infinite water softlock with two non-consecutive frame perfect inputs of rooster, and get rooster onto another screen, essentially working as a drop-in replacement for jesus jumps. it's not as consistent, but requires around the same amount of work, and allows longer jesus sections in about a third of the time it takes to jesus jump. i'm thinking of putting these in glitched logic?

### pros:
- only needs rooster.
- only need one per screen transition.
- inputs are not required to be consecutive, you can mash until you get the frames

### cons:
- upwards screen transitions might be impossible. i haven't been able to pull it off.
- annoying to hit the rooster frames out of water.
- two frame perfect inputs required.

### observations
- rooster doesnt splash, only link does.
- rooster avoids walls: this can persuade it back into link if it's 2-3 tiles away from the screen transition on a throw-after-pickup (aka "failed" attempt).
- risk is the same as multi-screen (feather) jesus jumping: fail = softlock
- rooster's jump position matters: it has a slower cycle than permadrowning, and they line up every  ~2-6 drowns, for ~1-2 drowns.

### how to perform:
1. use rooster to get to a screen edge
2. let go of rooster, buffer to enter the screen transition
3. you will be in the traditional water softlock at this point: mash rooster until you hit the right frame. rooster should appear when you get it, hopping over you.
4. mash rooster again until you actually grab it: BE EXTRA CAREFUL NOT TO THROW ROOSTER AFTER YOU GRAB IT. It is possible to ACTUALLY SOFTLOCK, as rooster can end up out of your reach.
5. repeat from step 1, across screens, until you get to your destination.

### where is this useful?
- below martha's bay cliff <-> animal village lower ledge near d5 (2 transitions)
- right of martha's bay cliff <-> below martha's bay cliff (2)
- d4 entrance/ledge above d4 <-> below left taltal (4)
- d4 entrance/ledge above d4 <-> mambo (1)
- d4 entrance/ledge above d4 <-> damp cave (3)
- bush near d3 <-> ukuku prairie (1)
- left taltal waterfall <-> below left taltal (3)
- above d6 maze <-> raft return (1)
- above d6 maze --> armos island (1)
- armos island <-> ukuku prairie (2)
- right of kanalet castle --> ukuku prairie (4)
- martha's bay mad batter top --> martha's bay mad batter bottom
- basically any jesus jump spot, but easier to pull off over a longer distance.

**Comment by foxsouns** [comment]
[jesus_rooster_marthas_bay_to_animal_village.webm](https://user-images.githubusercontent.com/60267141/212160204-4ab10267-481c-43a0-aa3a-e4a23a23db52.webm)

**Comment by daid** [comment]
If it's not consistent, then it sounds more like something for hell logic.

**Comment by foxsouns** [comment]
> If it's not consistent, then it sounds more like something for hell logic.

You know what: yeah, it's not exactly that "fun" to do. I'll move it over to hell logic.

**Comment by foxsouns** [comment]
ideas for making this potentially consistent:
___
- figuring out (if possible/how to) pause buffer rooster without bothering with drowning in the first place (like normal jesus jumps, where you buffer the item after landing in the water)
- s+q menu buffering an item after drowning (works for the first rooster pull, second one is still rooster position-dependent) [which is to buffer 3 s+q, then buffer rooster to pull]

**Comment by foxsouns** [comment]
@muffinjets suggested buffering the screen transition with a s+q menu input. That gets the first input reliable, but the second one is still problematic, as usual.
